### PR TITLE
Added Key method for kafka message

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- Added key for kafka message in order to send the message to specific Kafka partition.
- The associated tests are not included in this PR, but they will be added to the next PR with consumer parts.
- The blockNumber field is added for blockgroup result, so it is more consistent with the tracegroup result and it is used as a key for Kafka message.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: datasync/chaindatafetcher/kafka/repository.go [hunk 1]
```diff
@@ -26,11 +26,24 @@ import (
 	"github.com/klaytn/klaytn/datasync/chaindatafetcher/types"
 )
 
-type blockTraceResult struct {
+type traceGroupResult struct {
 	BlockNumber      *big.Int              `json:"blockNumber"`
 	InternalTxTraces []*vm.InternalTxTrace `json:"result"`
```
Comment (ehnuje) [inline datasync/chaindatafetcher/kafka/repository.go:31]: `json:"result"` for `InternalTxTraces`?

**Review by ehnuje — COMMENTED** [review]
Not need to be done in this PR, however, only few functions and types have comments. It would be better if descriptions are given to the functions and types.

File: datasync/chaindatafetcher/kafka/repository.go (hunk 1 above)
Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/kafka/repository.go:31]: Yes, it is the result of a block tracing.

**Comment by jeongkyun-oh** [comment]
> Not need to be done in this PR, however, only few functions and types have comments. It would be better if descriptions are given to the functions and types.

Alright. I will add more descriptions for better understanding.

**Review by KimKyungup — APPROVED** [review]

**Review by ehnuje — APPROVED** [review]

File: datasync/chaindatafetcher/kafka/kafka.go [hunk 2]
```diff
@@ -120,9 +125,13 @@ func (k *Kafka) Publish(topic string, data interface{}) error {
 	if err != nil {
 		return err
 	}
+	key := ""
+	if v, ok := data.(IKey); ok {
+		key = v.Key()
+	}
```
Comment (aidan-kwon) [inline datasync/chaindatafetcher/kafka/kafka.go:131]: Is there a valid case when `key == ""`?

Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/kafka/kafka.go:131]: If the key is not specified (""), the message will be inserted in the Round-Robin way.

**Review by yoomee1313 — APPROVED** [review]

**Comment by jeongkyun-oh** [comment]
@aidan-kwon @winnie-byun Please take a look.

**Review by winnie-byun — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Pull request merged**